base_app/base.py

```diff
@@ -18,53 +11,38 @@ def home():
 
 @app.route('/api/v1/recommendations', methods=['GET'])
 def index():
-    YELP_API_KEY = os.getenv('YELP_API_KEY')
-    headers = {'authorization': 'Bearer ' + YELP_API_KEY}
-    lat = request.args['latitude']
-    long = request.args['longitude']
-    url = 'https://api.yelp.com/v3/businesses/search'
-    if 'price' in request.args.keys():
-        price = request.args['price']
-        params = {'latitude': f'{lat}', 'longitude': f'{long}', 'price': f'{price}'}
-    elif 'categories' in request.args.keys():
-        cuisine = request.args['categories']
-        params = {'latitude': f'{lat}', 'longitude': f'{long}', 'term': f'{cuisine}'}
-    response = requests.get(url, params=params, headers=headers)
-    json_data = json.dumps(response.json())
-    restaurants = Restaurant.from_json(json_data)
-    restaurant = random.choice(restaurants)
-    schema = RestaurantSchema()
-    result = schema.dump(restaurant)
+    service = YelpService()
+    params = format_params(request.args)
+    return service.get_recommendation(params)
 
-    return result
 
 @app.route('/api/v1/photos', methods=['GET'])
-def get_photos():
-    YELP_API_KEY = os.getenv('YELP_API_KEY')
-    headers = {'authorization': 'Bearer ' + YELP_API_KEY}
-    lat = request.args['latitude']
-    long = request.args['longitude']
-    cuisine = request.args['categories']
-    params = {'latitude': f'{lat}', 'longitude': f'{long}', 'term': f'{cuisine}'}
-    url = 'https://api.yelp.com/v3/businesses/search'
+def show():
+    service = YelpService()
+    params = format_params(request.args)
+    restaurant_result = service.get_recommendation(params)
+    return service.get_photos(restaurant_result)
+
 
-    response = requests.get(url, params=params, headers=headers)
-    json_data = json.dumps(response.json())
-    restaurants = Restaurant.from_json(json_data)
-    restaurant = random.choice(restaurants)
-    schema = RestaurantSchema()
-    result = schema.dump(restaurant)
+## Helper Methods ##
 
-    restaurant_id = result['data']['id']
-    restaurant_url = f'https://api.yelp.com/v3/businesses/{restaurant_id}'
+def format_params(request_args):
```

Comment:
I like that you created this helper method to handle the various params. It definitely makes the route actions much simpler.